job/common/src/main/java/alluxio/job/plan/PlanConfig.java

```diff
@@ -9,22 +9,12 @@
  * See the NOTICE file distributed with this work for information regarding copyright ownership.
  */
 
-package alluxio.master.job;
+package alluxio.job.plan;
 
-import alluxio.proto.journal.Journal.JournalEntry;
+import alluxio.job.JobConfig;
 
 /**
- * An interface for a class which writes journal entries.
+ * A plan configuration.
  */
-public interface JournalEntryWriter {
-
-  /**
-   * Writes an entry to a checkpoint file.
-   *
-   * The entry should not have its sequence number set. This method will add the proper sequence
-   * number to the passed in entry.
-   *
-   * @param entry an entry to write to a journal checkpoint file
-   */
-  void writeJournalEntry(JournalEntry entry);
+public interface PlanConfig extends JobConfig {
```

Comment:
Do we plan to have something here in the future?

Reply:
This is what I currently use to differentiate between using the PlanTracker versus the WorkflowTracker in the JobMaster.

I think code snippet of what it looks like will help:


```
  public synchronized void run(JobConfig jobConfig, long jobId)
      throws JobDoesNotExistException, ResourceExhaustedException {
    // This RPC service implementation triggers another RPC.
    // Run the implementation under forked context to avoid interference.
    // Then restore the current context at the end.
    Context forkedCtx = Context.current().fork();
    Context prevCtx = forkedCtx.attach();
    try {
      if (jobConfig instanceof PlanConfig) {
        mPlanTracker.run((PlanConfig) jobConfig, mCommandManager, mJobServerContext,
            getWorkerInfoList(), jobId);
        return;
      } else if (jobConfig instanceof WorkflowConfig) {
        mWorkflowTracker.run((WorkflowConfig) jobConfig, jobId);
        return;
      }
      throw new JobDoesNotExistException(
          ExceptionMessage.JOB_DEFINITION_DOES_NOT_EXIST.getMessage(jobConfig.getName()));
    } finally {
      forkedCtx.detach(prevCtx);
    }
  }
```

The Plan part is the same but the Workflow part looks something like:

```
  public void run(WorkflowConfig workflowConfig, long jobId)
      throws JobDoesNotExistException, ResourceExhaustedException {
    WorkflowExecutionFactory<WorkflowConfig> executionFactory =
        WorkflowExecutionFactoryRegistry.INSTANCE.getExecutionFactory(workflowConfig);

    WorkflowExecution workflowExecution = executionFactory.create(workflowConfig);

    mWorkflows.put(jobId, workflowExecution);

    // Some other code to start off the workflow
    ...
    ....
  }
```